gnucash/register/register-gnome/gnucash-item-edit.c

```diff
@@ -814,6 +814,14 @@ button_press_cb (GtkWidget *widget, GdkEventButton *event, gpointer *pointer)
         g_signal_emit_by_name (sheet->reg, "show_popup_menu");
         return TRUE;
     }
+
+    // Middle mouse button, paste
+    if (event->button == 2 && event->type == GDK_BUTTON_PRESS)
+    {
+        gnc_item_edit_paste_clipboard (GNC_ITEM_EDIT(sheet->item_editor));
+        return TRUE;
+    }
+
```

Comment:
Middle mouse button paste is a default feature of a GtkEntry widget. You don't have to explicitly do this. It's one of the examples of where we were reimplementing standard Gtk behaviour.

Reply:
In the current setup we do, I think this is as a result of setting the entry so it can not be focused.